snmp/datadog_checks/snmp/config.py

```diff
@@ -180,14 +180,24 @@ def parse_metrics(self, metrics, enforce_constraints, warning, log):
         raw_oids = []
         table_oids = {}
         mibs_to_load = set()
+        loaded_table = {}
+
+        def get_table_symbols(mib, table):
+            key = (mib, table)
+            if key in loaded_table:
+                table_object = loaded_table[key]
+            else:
+                table_object = hlapi.ObjectType(hlapi.ObjectIdentity(mib, table))
+                loaded_table[key] = table_object
+                table_oids[table_object] = []
+            return table_oids[table_object]
```

Comment:
This implementation is ok, but I feel there is maybe a better way.

We are using here two data structures `table_oids` and `loaded_table`. I think we can probably find a way to use only one. But I don't see a solution that doesn't involve a lot of refactoring.

So, this is just a comment the point that, no need to necessarily address now :)